test/tap/tests/unit/protocol_unit-t.cpp

```
@@ -0,0 +1,418 @@
+	buf[2] = 0xFF;
+	bytes = mysql_decode_length(buf, &len);
+	ok(len == 0xFFFF && bytes == 3,
+		"decode_length: 0xFC 0xFF 0xFF → 65535 (3 bytes total)");
+}
+
+/**
+ * @brief Test mysql_decode_length() for 3-byte values (0xFD prefix).
+ */
+static void test_decode_length_3byte() {
+	unsigned char buf[4];
+	uint32_t len = 0;
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)